src/abstractdataframe/show.jl

```diff
@@ -573,61 +573,193 @@ function _show(io::IO,
+    num_rows, num_cols = size(df)
+
+    if !allcols
+        # Given the spacing, there is no way to fit more than W/9 rows of
+        # floating numbers in the screen, where W is the display width.
+        Δc = clamp(div(dsize[2], 9), 0, num_cols)
     else
-        rowindices1 = 1:bound
-        rowindices2 = max(bound + 1, nrows - nrowssubset + 1):nrows
+        Δc = num_cols
     end
-    maxwidths = getmaxwidths(df, io, rowindices1, rowindices2, rowlabel, rowid,
-                             eltypes, buffer, truncstring)
-    width = getprintedwidth(maxwidths)
-    showrows(io, df, rowindices1, rowindices2, maxwidths, splitcols, allcols,
-             rowlabel, summary, eltypes, rowid, buffer, truncstring)
-    return
+
+    if !allrows
+        # Get the maximum number of lines that we can display given the screen
+        # size.
+        Δr = clamp(dsize[1] - 4, 0, num_rows)
+    else
+        Δr = num_rows
+    end
+
+    Δr_lim = cld(Δr, 2)
+
+    # Do not align the numbers if there are more than 500 rows.
+    if Δr ≤ 500
+        @inbounds for i = 1:Δc
+            # TODO: Should we add support to `Union{Nothing, Float}`?
+
+            # Analyze the order of the number to compute the maximum padding
+            # that must be applied to align the numbers at the decimal point.
+            if nonmissingtype(types[i]) <: AbstractFloat
+                max_pad_i = 0
+                order_i   = zeros(Δr)
+                indices_i = zeros(Δr)
+
+                for k = 1:Δr
+                    # We need to process the top and bottom of the table because
+                    # we are cropping in the middle.
+
+                    if k ≤ Δr_lim
+                        kr = k
+                    else
+                        kr = num_rows - (k - Δr_lim) + 1
+                    end
+
+                    v = df[kr, i]
+
+                    order_v::Int = 0
+
+                    if v isa Number
+                        abs_v = abs(v)
+                        log_v::Int = (!isinf(v) && !isnan(v) && abs_v > 1) ? floor(Int, log10(abs_v)) : 0
+
+                        # If the order is higher than 5, then we print using
+                        # scientific notation.
+                        order_v = log_v > 5 ? 0 : floor(Int, log_v)
+
+                        # If the number is negative, we need to add an additional
+                        # padding to print the sign.
+                        v < 0 && (order_v += 1)
+                    end
+
+                    order_i[k]   = order_v
+                    indices_i[k] = kr
+
+                    order_v > max_pad_i && (max_pad_i = order_v)
+                end
+
+                push!(float_cols, i)
+                push!(indices, indices_i)
+                push!(padding, max_pad_i .- order_i)
+            end
+        end
+    end
+
+    # Create the formatter for floating point columns.
+    ft_float = (v, i, j)->_pretty_tables_float_formatter(v, i, j, float_cols,
```

Comment:
why anonymous function and not just an inner function?

Reply:
I am not sure if I understood. I need to create a new function using this closure to pass the variables from `float_cols` to `compact_printing` to the formatter. In PrettyTables.jl, the formatter API must have only three variables `v, i, j`.

Reply:
but you use anonymous function:
```
f = x -> x
```
instead of just standard function with a name
```
f(x) = x
```
and I am not clear why

Reply:
Ah! I am just used to this, should I change? Is it the second version better?

Reply:
for sure it is not worse. `f(x) = x` defines a `const`, while `f = x - > x` can be rebound later, so compiler has an easier life with `f(x) = x` to optimize things (but if it can prove that `f = x -> x` is not rebound later it is probably the same).

Reply:
Ok! I will change this.